src/ComfortPay/Message/ChargeRequest.php

```diff
@@ -18,79 +24,163 @@ public function setTerminalId($value)
+    {
+        return $this->setParameter('alpha2CountryCode', $value);
     }
 
     public function getData()
     {
-        $this->validate('terminalId', 'ws', 'amount', 'currency', 'transactionId', 'cid');
+        $this->validate('terminalId', 'amount', 'transactionId', 'transactionType', 'referedCardId', 'mid', 'currency');
 
         $data = parent::getData();
         $data = array_merge($data, [
-            'terminalId' => $this->getTerminalId(),
-            'ws' => $this->getWs(),
-            'cid' => $this->getCid(),
+            'transactionType' => $this->getTransactionType(),
             'transactionId' => $this->getTransactionId(),
+            'parentTransactionId' => $this->getParentTransactionId(),
+            'referedCardId' => $this->getReferedCardId(),
+            'merchantId' => $this->getMid(),
+            'terminalId' => $this->getTerminalId(),
             'amount' => $this->getAmount(),
-            'currency' => (new ISOCurrencies())->numericCodeFor(new Currency($this->getCurrency())),
+            'cc' => $this->getCurrency(),
+            'ipspData' => $this->getIpspData(),
             'vs' => $this->getVs(),
             'ss' => $this->getSs(),
+            'e2eReference' => $this->getE2eReference(),
+            'submerchantId' => $this->getSubmerchantId(),
+            'location' => $this->getLocation(),
+            'city' => $this->getCity(),
+            'alpha2CountryCode' => $this->getAlpha2CountryCode(),
         ]);
-        
+
         return $data;
     }
 
     public function sendData($data)
     {
-        if ($this->getTestmode()) {
+        if ($this->getTestmode())
+        {
             if (intval($data['cid']) % 2 == 0) {
-                return $this->response = new ChargeResponse($this, ['transactionStatus' => '02', 'transactionApproval' => '123']);
+                return $this->response = new ChargeResponse($this,
+                    ['transactionId' => $data['transactionId'], 'transactionStatus' => '02', 'transactionApproval' => '123']);
             }
-            return $this->response = new ChargeResponse($this, ['transactionStatus' => '00', 'transactionApproval' => '123']);
+            return $this->response = new ChargeResponse($this,
+                ['transactionId' => $data['transactionId'], 'transactionStatus' => '00', 'transactionApproval' => '123']);
+        }
+
+        $req = new \stdClass();
+        $req->transactionId = $data['transactionId'];
+        $req->referedCardId = $data['referedCardId'];
+        $req->merchantId = $data['merchantId'];
+        $req->terminalId = $data['terminalId'];
+        $req->amount = $data['amount'];
+        $req->parentTransactionId = $data['parentTransactionId'];
+        $req->cc = $data['cc'];
+
+        $symbols = new \stdClass();
+        $symbols->variableSymbol = $data['vs'];
+        $symbols->specificSymbol = $data['ss'];
+
+        $transactionIdentificator = new \stdClass();
```

Comment:
Based on the docs, `transactionIdentificator` should contain **either** `symbols` or `e2eReference`. Not both of them. You should pick one (I'd go with `e2eReference`) and if it's not present, use the other one.

If none of them are present, return validation error.